src/contexts/Transaction/TxHistoryProvider.tsx

```diff
@@ -22,18 +22,25 @@ const pollTransaction = async (
   if (!txLog.tx?.hash || txLog.status !== TxStatus.pending) {
     return txLog
   }
-
   const response = await readProvider.getTransaction(txLog.tx.hash)
-  // if no response, then the tx is cancelled.
+  const oneMinuteAgo = nowSeconds() - 1 * 60
   if (!response) {
-    txLog.callbacks?.onCancelled?.(response)
+    // if no response and tx was created over a minute ago, then the tx is cancelled.
```

Comment:
maybe just say here "we assume it's failed". Cos in theory i guess it might not actually have failed.